### PR TITLE
Qemu refactor travis experiment

### PR DESCRIPTION
**Pull request by richiejp**
Experiment with running tests inside a docker image: see #942

### CODE REVIEW

**Comment by coolo** [comment]
Why not pull @dasantiago 's image?

**Comment by richiejp** [comment]
It is an unnecessary external dependency from someone's personal repository. If the required dependencies changes then the image will need to be updated independently of the source. It probably makes sense to cache the image somehow, but I think a better solution is required.

**Comment by coolo** [comment]
It makes sense to use the same image openQA uses and not duplicate this. Ignore that it's hosted in a personal account for now.

**Comment by richiejp** [comment]
There are other problems as well, the openQA image needs a lot of work. Mine is much simpler, so it wouldn't take much to update or replace it. It also appears to be working.

File: docker/travis_test/prepare.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ zypper in -y -C \
        qemu-tools \
        qemu-kvm \
        tar \
+       opencv-devel \
```
Comment (coolo) [inline docker/travis_test/prepare.sh]: as much as you enjoy duplicating efforts - I will not merge this. We will only have one docker image

**Comment by richiejp** [comment]
Well, whatever Perl critic found some problems in the QEMU refactor so I will go fix those for now.

**Pull request closed without merging**